### ISSUE TITLE
DOC202 false positive when function doesn't return

### ISSUE THREAD

**Issue by NeilGirdhar** [astral-sh/ruff#12650]
```python
class C:
    def f(self) -> int:
        """Calculate x.

        Returns:
            x
        """
        raise NotImplementedError
```
Raises DOC202

**Comment by charliermarsh** [comment]
Thanks! Makes sense.
